lib/rouge/lexers/bibtex.rb

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+# regex based on https://github.com/SaswatPadhi/prismjs-bibtex and https://github.com/alecthomas/chroma/blob/master/lexers/b/bibtex.go
+
+module Rouge
+  module Lexers
+    class BibTeX < RegexLexer
+      title 'BibTeX'
+      desc "BibTeX"
+      tag 'bibtex'
+      filenames '*.bib'
+
+      valid_punctuation = /@\!\$\&\*\+\-\.\/\:\;\<\>\?\[\\\]\^\`\|\~/
```

Comment:
Stuck this here to make it a little easier to read. There's some memory hit to creating an additional object but given it's just one object it seemed worth it for clarity.